props/saber_BC_buttons.h

```diff
@@ -1267,28 +1288,43 @@ struct BCVolumeMode : public SPEC::SteppedMode {
       if (current_volume_ <= min_volume_) {
         current_volume_ = min_volume_;
         QuickMinVolume();
-      } else {
+      } else if (millis() - trigger_time_ > 100 && !sound_library_.busy()) {
         mode::getSL<SPEC>()->SayVolumeDown();
+        rangeState_ = RangeState::None;
+        trigger_time_ = millis();
       }
       dynamic_mixer.set_volume(current_volume_);
     }
   }
 
   void QuickMaxVolume() {
     dynamic_mixer.set_volume(max_volume_);
+    // Always play immediately when Max reached
+    if (rangeState_ == RangeState::SayingMin || rangeState_ == RangeState::SayingMax && 
+        sound_library_.busy()) {
+      sound_library_.fadeout(SaberBase::sound_length);
+    }
     PVLOG_NORMAL << "** Maximum Volume\n";
     mode::getSL<SPEC>()->SayMaximumVolume();
+    rangeState_ = RangeState::SayingMax;
   }
 
   void QuickMinVolume() {
     dynamic_mixer.set_volume(min_volume_);
+    if (rangeState_ == RangeState::SayingMin || rangeState_ == RangeState::SayingMax && 
```

Comment:
It seems very unlikely that rangeState_ would be RangeState::SayingMax when you get here. Like, that would only happen when you skip from max to min without any of the steps in between.

It also seems like if it's SayingMin, we should just let it finish rather than fading it out?

To me, it seems like the logic should be:

```
   if (rangeState_ == RangeState::SayingMin && sound_ibrary_.busy()) return;
   sound_library_.fadeout(0.1);
```